### PR TITLE
ENG-17059: Basic tests for file, http and jdbc export clients.

### PR DESCRIPTION
**Pull request by manjujames**

### CODE REVIEW

File: tests/frontend/org/voltdb/export/TestExportClientBasic.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        int actualRows = 0;
+        for(Map.Entry<String, String> entry : cluster.getHostRoots().entrySet()) {
+            File voltdbRoot = new File(entry.getValue(), FILE_EXPORT_DIR);
+            for (File file : voltdbRoot.listFiles()) {
+                if (file.isFile()) {
+                    actualRows += countRowsInFile(file);
+                }
+            }
+        }
+        assertEquals(expectedNumRows, actualRows);
```
Comment (rdykiel) [inline tests/frontend/org/voltdb/export/TestExportClientBasic.java]: Note: in cases where we will test node rejoins, we will end up with actualRows >= expectedRows. We should think about checking that we got all the sequence numbers at least once (requires parsing each row).

File: tests/frontend/org/voltdb/regressionsuites/LocalCluster.java [hunk 2]
```diff
@@ -2311,6 +2311,16 @@ private static LocalCluster compileBuilder(
         return lc;
     }
 
+    private ClientConfig createClientConfig() {
+        ClientConfig cc = new ClientConfig();
+        cc.setProcedureCallTimeout(10 * 60 * 1000);
```
Comment (n3world) [inline tests/frontend/org/voltdb/regressionsuites/LocalCluster.java]: 10 minutes seems like a very long timeout especially for a test

File: tests/frontend/org/voltdb/export/TestExportClientBasic.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            + "PARTITION ON COLUMN col1 "
+            + "EXPORT TO TARGET s1_target ("
+            + "     col1 INTEGER NOT NULL, "
+            + "     col2 INTEGER NOT NULL"
+            + ");";
+    
+    private static final String JDBC_URL = "jdbc:hsqldb:mem:x1";
+    private static final String JDBC_USER = "sa";
+    private static final String HSQL_DRIVER = "org.hsqldb_voltpatches.jdbcDriver";
+    private static Map<String, ExportConfig> s_exportConfigs = new HashMap<>();
```
Comment (n3world) [inline tests/frontend/org/voltdb/export/TestExportClientBasic.java]: You could also use Parameterized test to iterate over arguments passed to the constructor. The TestDR2Truncate suite uses this feature.
https://github.com/junit-team/junit4/wiki/parameterized-tests

File: src/frontend/org/voltdb/exportclient/JDBCExportClient.java [hunk 4]
```diff
@@ -379,6 +379,10 @@ private void initialize(long generation, String stableName, List<String> columnN
                 identifierQuoteTemp = "";
                 supportsUpsert = true;
                 m_disableAutoCommits = false;
+            } else if (dbName.contains("HSQL")) {
```
Comment (n3world) [inline src/frontend/org/voltdb/exportclient/JDBCExportClient.java]: Why is this using contains, all of the others are using equals?

Comment (manjujames) [reply src/frontend/org/voltdb/exportclient/JDBCExportClient.java]: The db name from HSQL is a longer name.

File: tests/frontend/org/voltdb/export/TestExportClientBasic.java (hunk 3 above)
Comment (manjujames) [reply tests/frontend/org/voltdb/export/TestExportClientBasic.java]: Thanks, I will look into that when adding more tests for the other export types.

File: tests/frontend/org/voltdb/export/TestExportClientBasic.java (hunk 1 above)
Comment (manjujames) [reply tests/frontend/org/voltdb/export/TestExportClientBasic.java]: I will take care of it when I add rejoin.

File: tests/frontend/org/voltdb/regressionsuites/LocalCluster.java (hunk 2 above)
Comment (manjujames) [reply tests/frontend/org/voltdb/regressionsuites/LocalCluster.java]: Copy-pasted without giving it much thought :). I will reduce it to 30 seconds or so.

File: tests/frontend/org/voltdb/export/TestExportClientBasic.java [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+    private enum TestType {
+        FILE,
+        HTTP,
+        JDBC,
+        KAFKA;
+    }
+
+    @Parameterized.Parameters
+    public static Collection<ExportConfig> exportConfigs() {
+       return Arrays.asList(new ExportConfig[] {
```
Comment (n3world) [inline tests/frontend/org/voltdb/export/TestExportClientBasic.java]: Arrays.asList is varargs so you shouldn't have to have the `new ExportConfig[] {`

**Pull request merged**